modules/RTC/TPCUtils.js

```diff
@@ -0,0 +1,365 @@
+                    this.pc.eventEmitter.emit(RTCEvents.LOCAL_TRACK_SSRC_UPDATED,
+                        newTrack,
+                        this.pc._extractPrimarySSRC(ssrc));
+                });
+        } else if (oldTrack && !newTrack) {
+            if (!this.removeTrackMute(oldTrack)) {
+                return Promise.reject(new Error('replace track failed'));
+            }
+            this.pc.localTracks.delete(oldTrack.rtcId);
+            this.pc.localSSRCs.delete(oldTrack.rtcId);
+        } else if (newTrack && !oldTrack) {
+            const ssrc = this.pc.localSSRCs.get(newTrack.rtcId);
+
+            if (!this.addTrackUnmute(newTrack)) {
+                return Promise.reject(new Error('replace track failed'));
+            }
+            newTrack.emit(JitsiTrackEvents.TRACK_MUTE_CHANGED, newTrack);
+            this.pc.localTracks.set(newTrack.rtcId, newTrack);
+            this.pc.localSSRCs.set(newTrack.rtcId, ssrc);
+        }
+
+        return Promise.resolve(false);
+    }
+
+    /**
+     *
+     * @param {boolean} active
+     */
+    setAudioTransferActive(active) {
+        return this.setMediaTransferActive('audio', active);
+    }
+
+    /**
+     * Set the simulcast stream encoding properties on the RTCRtpSender.
+     * @param {*} track - the current track in use for which the encodings are to be set.
+     */
+    setEncodings(track) {
+        const transceiver = this.pc.peerconnection.getTransceivers()
+            .find(t => t.sender && t.sender.track && t.sender.track.kind === track.getType());
+        const parameters = transceiver.sender.getParameters();
+
+        parameters.encodings = this._getStreamEncodings(track);
+        transceiver.sender.setParameters(parameters);
+    }
+
+    /**
+     *
+     * @param {*} mediaType
+     * @param {boolean} active
+     */
+    setMediaTransferActive(mediaType, active) {
+        const transceivers = this.pc.peerconnection.getTransceivers()
+            .filter(t => t.receiver && t.receiver.track && t.receiver.track.kind === mediaType);
+
+        if (active) {
+            transceivers.forEach(transceiver => {
+                if (this._getLocalTracks(mediaType).length > 0) {
+                    transceiver.direction = 'sendrecv';
+                    const parameters = transceiver.sender.getParameters();
+
+                    if (parameters && parameters.encodings && parameters.encodings.length) {
+                        parameters.encodings.forEach(encoding => {
+                            encoding.active = true;
+                        });
+                        transceiver.sender.setParameters(parameters);
+                    }
+                } else {
+                    transceiver.direction = 'recvonly';
+                }
+            });
+        } else {
+            transceivers.forEach(transceiver => {
+                transceiver.direction = 'inactive';
+            });
+        }
+
+        return true;
```

Comment:
this function always returns true

Reply:
thanks for pointing it out, this should be returning false always so that renegotiation is not triggered.